src/wasm-interpreter.h

```diff
@@ -4023,7 +4023,7 @@ class ModuleRunnerBase : public ExpressionRunner<SubType> {
     const auto& seg = *wasm.getDataSegment(curr->segment);
     auto elemBytes = element.getByteSize();
     auto end = offset + size * elemBytes;
-    if ((size != 0ull && droppedDataSegments.count(curr->segment)) ||
+    if ((offset + size > 0 && droppedDataSegments.count(curr->segment)) ||
```

Comment:
It looks like we should check for overflow as well.

Reply:
Done.